R/create_individual_file.R

```diff
@@ -0,0 +1,861 @@
+    ) %>%
+    dplyr::mutate(
+      ch_beddays = dplyr::if_else(
+        recid == "CH",
+        as.numeric(term_1 - term_2),
+        NA_real_
+      ),
+      ch_cost = dplyr::if_else(
+        recid == "CH" & ch_no_cost == 0,
+        ch_beddays * ch_cost_per_day,
+        NA_real_
+      ),
+      ch_beddays = dplyr::if_else(
+        recid == "CH" & first_ch_ep == 0,
+        0,
+        ch_beddays
+      ),
+      ch_cost = dplyr::if_else(
+        recid == "CH" & first_ch_ep == 0,
+        0,
+        ch_cost
+      )
+    ) %>%
+    dplyr::select(
+      -fy_end, -fy_start, -term_1, -term_2
+    )
+}
+
+#' Date from FY
+#'
+#' @description Return start, mid, or end date from financial year in format "2122".
+#'
+#' @param financial_year Financial year represented in "YYYY" format e.g. "2122"
+#' @param type One of "start", "end", and "mid", representing the date to return
+date_from_fy <- function(financial_year, type = c("start", "end", "mid")) {
+  match.arg(type)
+  n <- switch(type,
+    "start" = 0,
+    "mid" = 0,
+    "end" = 2
+  )
+  year <- as.numeric(paste0("20", substr(financial_year, 1 + n, 2 + n)))
+  if (type == "start") {
+    date <- lubridate::make_date(year, 4, 1)
+    return(date)
+  } else if (type == "end") {
+    date <- lubridate::make_date(year, 3, 31)
+    return(date)
+  }
+  date <- lubridate::make_date(year, 9, 30)
+  return(date)
+}
+
+#' Recode gender
+#'
+#' @description Recode gender to 1.5 if 0 or 9.
+#'
+#' @inheritParams create_individual_file
+recode_gender <- function(episode_file) {
+  episode_file %>%
+    dplyr::mutate(
+      gender = dplyr::if_else(
+        gender == 0 | gender == 9,
+        1.5,
+        gender
+      )
+    )
+}
+
+#' Aggregate by CHI
+#'
+#' @description Aggregate episode file by CHI to convert into
+#' individual file.
+#'
+#' @inheritParams create_individual_file
+aggregate_by_chi <- function(episode_file) {
+  episode_file %>%
+    dplyr::arrange(
+      chi,
+      record_keydate1,
+      keytime1,
+      record_keydate2,
+      keytime2
+    ) %>%
+    dplyr::group_by(chi) %>%
+    dplyr::summarise(
+      distinct_cij = n_distinct("cij_marker"),
+      ooh_cases = n_distinct("ooh_case_id"),
+      gender = mean(gender),
+      dplyr::across(dplyr::ends_with(c(
+        "postcode", "DoB", "gpprac"
+      )), ~ dplyr::last(., na_rm = TRUE)),
+      dplyr::across(
+        c(
+          "CIJ_el",
+          "CIJ_non_el",
+          "CIJ_mat",
+          # "cij_delay",
+          "OoH_cases" = "unique_ooh_case",
+          dplyr::ends_with(
+            c(
+              "episodes",
+              "beddays",
+              "cost",
+              "attendances",
+              "attend",
+              "contacts",
+              "hours",
+              "alarms",
+              "telecare",
+              "paid_items",
+              "advice",
+              "homeV",
+              "time",
+              "admissions",
+              "assesment",
```

Comment:
Same typo as with `[ooh_covid_assessment](https://github.com/Public-Health-Scotland/source-linkage-files/pull/561/commits/979fc81ae48510b15b096feaf86779064e9d4587)` and I think that change will break this bit.
```suggestion
              "assessment",
```